source/server/infrastructure/commit_publisher.coffee

```diff
@@ -29,37 +28,68 @@ class Space.eventSourcing.CommitPublisher extends Space.Object
           update: $push: { receivers: { appId: appId, receivedAt: new Date() } }
         })
         # Only publish the event if this process was the one that locked it
-        @publishCommit(@_parseCommit(lockedCommit)) if lockedCommit?
+        @publishChanges(@parseChanges(lockedCommit), commit._id) if lockedCommit?
     }
+    @log.info @_logMsg("observing for commits via the collection")
 
   stopPublishing: ->
-    @log.info @_logMsg("stopped")
     @_publishHandle?.stop()
+    @log.info @_logMsg("collection observer stopped")
 
-  publishCommit: (commit) =>
-    @_setProcessingTimeout(commit)
+  publishChanges: (changes, commitId) =>
+    @_setTimeout(changes, commitId)
     try
-      for event in commit.changes.events
+      for event in changes.events
         @log.debug @_logMsg("publishing #{event.typeName()}"), event
         @eventBus.publish event
-      for command in commit.changes.commands
+      for command in changes.commands
         @log.debug @_logMsg("sending #{command.typeName()}"), command
         @commandBus.send command
     catch error
-      @_failCommitProcessingAttempt(commit._id)
-      throw new Error "while publishing:\n
-        #{JSON.stringify(commit)}\n
-        error:#{error.message}\n
-        stack:#{error.stack}"
-    @_markAsProcessed(commit)
-
-  _setProcessingTimeout: (commit) ->
-    @_inProgress[commit._id] = @meteor.setTimeout (=>
-      @log.error(@_logMsg("#{commit._id} timed out"), commit)
-      @_failCommitProcessingAttempt(commit._id)
+      @commits.update(
+        { _id: commitId, 'receivers.appId': @configuration.appId },
+        { $set: { 'receivers.$.failedAt': new Date() } }
+      )
+      throw error
```

Comment:
Just switched to throwing the same error, as if it get's this far we should probably let it crash the process without translating since we're not doing anything up the chain. The JSON string is pretty horrible too, so another reason to just re-throw. 
cc @meteor-space/core